### PR TITLE
Refactoring and windows support

### PR DESCRIPTION
**Pull request by ErikBjare**

### CODE REVIEW

File: aw_watcher_afk/windows.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    paramflags = (1, "lastinputinfo"),
+    c_GetLastInputInfo = prototype(("GetLastInputInfo", ctypes.windll.user32), paramflags)
+
+    l = LastInputInfo()
+    l.cbSize = ctypes.sizeof(LastInputInfo)
+    assert 0 != c_GetLastInputInfo(l)
+    return l.dwTime
+
+
+def _getTickCount() -> int:
```
Comment (ErikBjare) [inline aw_watcher_afk/windows.py]: We should probably use [GetTickCount64](https://msdn.microsoft.com/en-us/library/windows/desktop/ms724411(v=vs.85).aspx) instead.

File: aw_watcher_afk/windows.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    prototype = WINFUNCTYPE(DWORD)
+    paramflags = ()
+    c_GetTickCount = prototype(("GetTickCount", ctypes.windll.kernel32), paramflags)
+    return c_GetTickCount()
+
+
+def time_since_last_input():
```
Comment (ErikBjare) [inline aw_watcher_afk/windows.py]: Note that `getTickCount` and `getLastInputTick` both only have DWORD (32bit uint) length and will wrap after that. We might want to check for negative values or something and correct as necessary.

File: aw_watcher_afk/afk.py [hunk 3]
```diff
@@ -14,193 +14,146 @@
...
-    """ Set up argparse """
-    parser = argparse.ArgumentParser("A watcher for keyboard and mouse input to detect AFK state")
-    parser.add_argument("-v", dest='verbose', action="store_true",
-                        help='run with verbose logging')
-    parser.add_argument("--testing", action="store_true",
-                        help='run in testing mode')
-    parser.add_argument("--desktop-notify", action="store_true",
-                        help='sends desktop notifications when you become afk/non-afk')
-    args = parser.parse_args()
+class AFKWatcher:
```
Comment (johan-bjareholt) [inline aw_watcher_afk/afk.py]: Now when we have a class for it, wouldn't it also make sense to move it to its own file aswell?

Comment (ErikBjare) [inline aw_watcher_afk/afk.py]: Done.

**Comment by ErikBjare** [comment]
Not so WIP anymore, please try it out @johan-bjareholt!

File: aw_watcher_afk/windows.py (hunk 2 above)
Comment (ErikBjare) [inline aw_watcher_afk/windows.py]: Created an issue for this: #16

**Comment by johan-bjareholt** [comment]
I don't have access to a windows comuputer currently but tested it on linux. Got this crash everytime i became AFK

```
2017-04-22 12:09:47,396 [INFO ]: aw-server destination set to localhost:5666 (client.py:43)
2017-04-22 12:09:47,398 [INFO ]: afkwatcher started (afk.py:92)
2017-04-22 12:09:47,438 [INFO ]: Connection to aw-server established (client.py:224)
2017-04-22 12:09:48,521 [INFO ]: No longer AFK (afk.py:122)
2017-04-22 12:10:22,564 [INFO ]: Became AFK (afk.py:125)
Traceback (most recent call last):
  File "/usr/bin/aw-watcher-afk", line 11, in <module>
    load_entry_point('aw-watcher-afk', 'console_scripts', 'aw-watcher-afk')()
  File "/home/johan/Programming/activitywatch/aw-watcher-afk/aw_watcher_afk/__main__.py", line 28, in main
    watcher.run()
  File "/home/johan/Programming/activitywatch/aw-watcher-afk/aw_watcher_afk/afk.py", line 126, in run
    self.change_state(self.last_activity)
AttributeError: 'AFKWatcher' object has no attribute 'last_activity'
```

Fixed it in last commit by setting self.last_activity again, seems to work fine now.

**Comment by ErikBjare** [comment]
@johan-bjareholt Ah yeah, I got that too but forgot to commit the fix.

File: aw_watcher_afk/afk.py [hunk 4]
```diff
@@ -116,6 +117,7 @@ def run(self):
                     self.afk = False
                 else:
                     time_since_last_input = get_time_since_last_input()
+                    self.second_last_activity = self.last_activity
```
Comment (ErikBjare) [inline aw_watcher_afk/afk.py]: I'm not convinced this will have the right behavior...

**Comment by ErikBjare** [comment]
@johan-bjareholt So, does this work alright for you?

**Comment by johan-bjareholt** [comment]
@ErikBjare Works fine on linux yes, have still not have time to test on windows though.

**Comment by ErikBjare** [comment]
@johan-bjareholt Not worried about Windows support. It's unlikely to have any issues. Merging.

**Pull request merged**